src/plugins/intel_gpu/tests/test_cases/reorg_yolo_gpu_test.cpp

```diff
@@ -0,0 +1,372 @@
+}
+
+struct PrintToStringParamName {
+    template<class T>
+    std::string operator()(const testing::TestParamInfo<ReorgYoloParamsWithLayout<T> > &param) {
+        std::stringstream buf;
+        ReorgYoloParams<T> p;
+        format::type target_format;
+        bool should_fail;
+        std::tie(p, target_format, should_fail) = param.param;
+        buf << "InputTensor=" << to_string(p.inputTensor)
+            << ".stride=" << p.stride
+            << ".TargetLayout=" << fmt_to_str(target_format);
+        return buf.str();
+    }
+};
+};  // namespace
+
+template<typename T>
+struct reorg_yolo_test
+        : public ::testing::TestWithParam<ReorgYoloParamsWithLayout<T> > {
+public:
+    void test() {
+        ReorgYoloParams<T> params;
+        format::type target_format;
+        bool should_fail;
+        std::tie(params, target_format, should_fail) = this->GetParam();
+
+        if (should_fail) {
+            ASSERT_THROW(run_test(params, target_format), std::invalid_argument);
+        } else {
+            ASSERT_NO_FATAL_FAILURE(run_test(params, target_format));
+        }
+    }
+
+private:
+    void run_test(const ReorgYoloParams<T>& params, const format::type target_format) {
+        const auto data_type = type_to_data_type<T>::value;
+        const format::type plain_format = format::bfyx;
+
+        auto& engine = get_test_engine();
+
+        auto input = engine.allocate_memory({params.inputTensor, data_type, plain_format});
+
+        set_values(input, params.input);
+
+        topology topology;
+        topology.add(input_layout("input", input->get_layout()));
+        topology.add(reorder("input_reordered", "input", target_format, data_type));
+        topology.add(reorg_yolo("reorg_yolo", "input_reordered", params.stride));
+        topology.add(reorder("reorg_yolo_reordered", "reorg_yolo", plain_format, data_type));
+
+        network network(engine, topology);
+        network.set_input_data("input", input);
+        const auto result = network.execute();
+
+        auto out_mem = result.at("reorg_yolo_reordered").get_memory();
+        cldnn::mem_lock<T> out_ptr(out_mem, get_test_stream());
+
+        ASSERT_EQ(params.expected.size(), out_ptr.size());
+        for (size_t i = 0; i < params.expected.size(); ++i) {
+            EXPECT_NEAR(params.expected[i], out_ptr[i], getError<T>()) << "format=" << target_format << ", i= " << i;
+        }
+    }
+};
+
+
+using test_f32 = reorg_yolo_test<float>;
+using test_f16 = reorg_yolo_test<half_t>;
+
+TEST_P(test_f32, basic) {
+    test();
+}
+
+TEST_P(test_f16, basic) {
+    test();
+}
+
+
+
+INSTANTIATE_TEST_SUITE_P(reorg_yolo_f32,
+                         test_f32,
+                         ::testing::Combine(
+                                 ::testing::ValuesIn(generateParams<float>()),
+                                 ::testing::ValuesIn(dataFormats),
+                                 ::testing::Values(false)),
+                         PrintToStringParamName());
```

Comment:
this will generate 7*3 test cases and I think we can reduce the test scope. I think it is ok to use only 1 from generateParam as others are tested from f16.
(general note: `combine` might generate too much test cases and it increases validation pressure. Let's not put too much combination..)

Reply:
Done.